File: Week_5_Data_structures/src/ListIntroduction1.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.ListIterator;

public class ListIntroduction1 {
    public static void main(String[] args) {

        //Create an empty list which will contain names (strings)
        ArrayList<String> names = new ArrayList<>();
        //Print out the number of elements in the list
        System.out.println(names.size());

        //Add William to the list
        names.add("William");
        //Print out whether the list is empty or not
        System.out.println(names.isEmpty());

        //Add John to the list
        names.add("John");
        //Add Amanda to the list
        names.add("Amanda");

        //Print out the number of elements in the list
        System.out.println(names.size());
        //Print out the 3rd element
        System.out.println(names.get(2));

        System.out.println();

        //Iterate through the list and print out each name
        Iterator indivNames = names.iterator();
        while(indivNames.hasNext()){
            System.out.println(indivNames.next());
        }

        //Iterate through the list and print
        ListIterator listNames = names.listIterator();
        for (int i = 0; i < names.size(); i++) {
            System.out.printf("%d. %s\n",i + 1, names.get(i));
        }

        System.out.println();

         //Remove the 2nd element
         names.remove(1);
         //Iterate through the list in a reversed order and print out each name
            Iterator removedNamesList = names.iterator();
            while(removedNamesList.hasNext()) {
                System.out.println(removedNamesList.next());
            }

         //Remove all elements
        names.clear();

        System.out.println();

        System.out.println(names.isEmpty());

    }
}

File: Project_phase_FE/DataStructures/src/ListIntroduction1.js
var names = [];

// Print out the number of elements in the list
console.log(names.length);

// Add William to the list
names.push("William");
console.log(...names);

// Print out whether the list is empty or not
if (names.length < 1){
    console.log("The names list is empty");
} else {
    console.log("The names list size is: " + names.length);
}

// Add John to the list
names.push("John");

// Add Amanda to the list
names.push("Amanda");

// Print out the number of elements in the list
console.log(names.length);

// Print out the 3rd element
console.log(names[2]);

// Iterate through the list and print out each name
for(var i = 0; i < names.length; i++){
    console.log(names[i]);
}

// Iterate through the list and print
for(var i = 0; i < names.length; i++){
    console.log((i + 1) + ". " + names[i]);
}

// Remove the 2nd element
function remove(elementNumber){
    var removeName = names.indexOf(elementNumber);
    return removeName;
}
remove(2);
names.pop();

// Iterate through the list in a reversed order and print out each name
for(var i = names.length - 1; i >= 0; i--){
    console.log(names[i]);
}

// Remove all elements
names.length = 0;
console.log(names);


